pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -36,12 +36,27 @@ class ListAppend(PyccelInternalFunction):
     """
     __slots__ = ()
     _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = -1
+    _class_type = NativeHomogeneousList()
     name = 'append'
 
-    def __init__(self, *args) -> None:
-        super().__init__(*args)
-        if not isinstance(args[1].dtype, type(args[0].dtype)):
-            errors.report(f'Argument of type {args[1].dtype} is incompatible with list of type {args[0].dtype}',
-                        severity='fatal')
-
-
+    def __init__(self, lst_bound_arg, new_elem) -> None:
+        super().__init__(new_elem)
+        list_precision = get_final_precision(lst_bound_arg)
+        conditions = [
+            new_elem.dtype is not NativeGeneric() and
+            lst_bound_arg.dtype == new_elem.dtype and
+            list_precision == get_final_precision(new_elem) and
+            lst_bound_arg.rank - 1 == new_elem.rank and
+            (getattr(lst_bound_arg, 'order', None) == getattr(new_elem, 'order',None))
+        ]
```

Comment:
This is a list containing 1 boolean. If you use () then you can avoid the use of `all` in the line below.
```suggestion
        conditions = (
            new_elem.dtype is not NativeGeneric() and
            lst_bound_arg.dtype == new_elem.dtype and
            list_precision == get_final_precision(new_elem) and
            lst_bound_arg.rank - 1 == new_elem.rank and
            (getattr(lst_bound_arg, 'order', None) == getattr(new_elem, 'order',None))
        )
```